targets/mbedos5/jerryscript-mbed/jerryscript-mbed-event-loop/EventLoop.h

```diff
@@ -0,0 +1,105 @@
+/* Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _JERRYSCRIPT_MBED_EVENT_LOOP_EVENT_LOOP_H
+#define _JERRYSCRIPT_MBED_EVENT_LOOP_EVENT_LOOP_H
+
+#include "jerry-core/jerry-api.h"
+
+#include "Callback.h"
+#include "mbed-events/EventQueue.h"
+
+#include "jerryscript-mbed-util/logging.h"
+#include "jerryscript-mbed-event-loop/BoundCallback.h"
+
+#include <vector>
+
+extern "C" void exit(int return_code);
+
+namespace mbed {
+namespace js {
+
+static const uint32_t EVENT_INTERVAL_MS = 1;
+
+class EventLoop {
+ private:
+    static EventLoop instance;
+
+ public:
+    static EventLoop& getInstance() {
+        return instance;
+    }
+
+    void go() {
+        while (true) {
+            queue.dispatch();
+        }
+    }
+
+    Callback<void()> wrapFunction(jerry_value_t f) {
+        if (!jerry_value_is_function(f)) {
+            LOG_PRINT_ALWAYS("invalid function passed\r\n");
+            exit(1);
```

Comment:
Is this really what we want? That `exit` signals some bad smell to me. Shouldn't these four lines just be replaced with an `assert(jerry_value_is_function(f));`, or something similar?